airbyte-integrations/connectors/destination-postgres/src/main/java/io/airbyte/integrations/destination/postgres/PostgresDestination.java

```diff
@@ -271,23 +285,30 @@ public void close(boolean hasFailed) throws Exception {
 
         // write anything that is left in the buffers.
         writeStreamsWithNRecords(0, 500, writeConfigs, database);
+        Set<String> schemaSet = new HashSet<>();
 
         database.transaction(ctx -> {
           final StringBuilder query = new StringBuilder();
           for (final WriteConfig writeConfig : writeConfigs.values()) {
             // create tables if not exist.
-            query.append(createRawTableQuery(writeConfig.getTableName()));
+            final String schemaName = writeConfig.getSchemaName();
+            if (!schemaSet.contains(schemaName)) {
```

Comment:
why are we creating the schemas in two places in this class? lines 144 and 296. i think you can just do it on line 144 and be done with it?

Reply:
yes you're right